src/vivarium/component.py

```diff
@@ -0,0 +1,143 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, List, Optional
+
+from vivarium.framework.event import Event
+
+if TYPE_CHECKING:
+    from vivarium.framework.engine import Builder
+    from vivarium.framework.population import PopulationView, SimulantData
+
+
+class VivariumComponent(ABC):
+    """A component that can be used in a Vivarium simulation."""
+
+    configuration_defaults = {}
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """The name of the component. Names must be unique within a simulation."""
+        pass
+
+    @property
+    def sub_components(self) -> List["VivariumComponent"]:
+        """A list of the components that are managed by this component."""
+        return self._sub_components
+
+    @property
+    def columns_created(self) -> List[str]:
+        """A list of the columns this component creates."""
+        return []
+
+    @property
+    def columns_required(self) -> List[str]:
+        """A list of the columns this component requires that it did not create."""
+        return []
+
+    @property
+    def initialization_columns_required(self) -> List[str]:
+        """A list of the columns this component requires during simulant initialization."""
+        return []
+
+    @property
+    def time_step_prepare_priority(self) -> int:
+        return 5
```

Comment:
I just looked this up and in case anyone reading doesn't know - the event manager in vivarium defaults to priority 5 when registering listeners if not specified :)